test/L1ERC20Bridge.t.sol

```diff
@@ -3,33 +3,159 @@ pragma solidity ^0.8.0;
+    _withdraw(account, amount);
+  }
+
+  function receiveWithdrawalWormholeMessages(
+    bytes memory payload,
+    bytes[] memory additionalVaas,
+    bytes32 callerAddr,
+    uint16 sourceChain,
+    bytes32 deliveryHash
+  ) public {
+    _receiveWithdrawalWormholeMessages(
+      payload, additionalVaas, callerAddr, sourceChain, deliveryHash
+    );
+  }
+}
+
+contract L1ERC20BridgeTest is Constants, WormholeRelayerBasicTest {
+  L2ERC20 l2Erc20;
+  FakeERC20 fake;
+  L1ERC20Bridge bridge;
+
+  constructor() {
+    setTestnetForkChains(6, 5);
+  }
+
+  function setUpSource() public override {
+    fake = new FakeERC20("Hello", "WRLD");
+    IGovernor gov = new GovernorMock("Testington Dao", fake);
+    bridge =
+    new L1ERC20Bridge(address(fake), wormholeCoreFuji, address(gov), wormholeFujiId, wormholePolygonId);
+  }
+
+  function setUpTarget() public override {
+    L1Block l1Block = new L1Block();
+    l2Erc20 =
+    new L2ERC20( "Hello", "WRLD", wormholeCoreMumbai, address(l1Block), wormholePolygonId, wormholeFujiId);
+  }
+}
+
+contract Constructor is Test, Constants {
+  function testFork_CorrectlySetAllArgs(address l1Erc) public {
+    FakeERC20 fake = new FakeERC20("Hello", "WRLD");
+    IGovernor gov = new GovernorMock("Testington Dao", fake);
+    L1ERC20Bridge bridge =
+    new L1ERC20Bridge(address(l1Erc), wormholeCoreFuji, address(gov), wormholeFujiId, wormholePolygonId);
+    assertEq(address(bridge.L1_TOKEN()), l1Erc, "L1 token is not set correctly");
+  }
+}
+
+contract Initialize is L1ERC20BridgeTest {
+  function testFork_CorrectlyInitializeL2Token(address l2Erc20) public {
+    bridge.initialize(address(l2Erc20));
+    assertEq(bridge.L2_TOKEN_ADDRESS(), l2Erc20, "L2 token address is not setup correctly");
+    assertEq(bridge.INITIALIZED(), true, "Bridge isn't initialized");
+  }
+
+  function testFork_InitializeL2AddressWhenAlreadyInitialized(address l2Erc20) public {
+    bridge.initialize(address(l2Erc20));
+
+    vm.expectRevert(L1ERC20Bridge.AlreadyInitialized.selector);
+    bridge.initialize(address(l2Erc20));
   }
 }
 
 contract Deposit is L1ERC20BridgeTest {
-  function testFork_CorrectlyDepositTokens() public {
-    FakeERC20 erc20 = new FakeERC20("Hello", "WRLD");
-    IGovernor gov = new GovernorMock("Testington Dao", erc20);
-    L1ERC20Bridge bridge = new L1ERC20Bridge(address(erc20), wormholeCoreFuji, address(gov));
-    bridge.initialize(0xBaA85b5C4c74f53c46872acfF2750f512bcBEC43);
-
-    erc20.approve(address(bridge), 100_000);
-    erc20.mint(address(this), 100_000);
+  function testFork_CorrectlyDepositTokens(uint96 _amount) public {
+    bridge.initialize(address(l2Erc20));
+    uint256 cost = bridge.quoteDeliveryCost(wormholePolygonId);
+    vm.recordLogs();
+
+    fake.approve(address(bridge), _amount);
+    fake.mint(address(this), _amount);
+    vm.deal(address(this), 1 ether);
+
+    bridge.deposit{value: cost}(address(this), _amount);
```

Comment:
We should assert the L1 tokens have been transferred to the bridge contract.

Reply:
https://github.com/ScopeLift/l2-flexible-voting/pull/21/commits/a51b9d3189cf402f0b897e490bd028b36a38eaaa